### ISSUE TITLE
Passing `self` into child objects

### ISSUE THREAD

**Issue by jack-mcivor** [astral-sh/ruff#19456]
### Summary

I came across this blog that warns against unnecessary coupling caused by passing `self` into child objects https://johnscolaro.xyz/blog/python-code-smells-passing-self & thought this seems like a good lint rule.

An example is
```python
class Car:
    def __init__(self):
        self.speed = 0
        self.wheels = [Wheel(self) for _ in range(4)]  # BAD! passing-self

class Wheel:
    def __init__(self, car: Car): # Cursed Line
        self.wear = 0
        self.car = car # Cursed Line

    def check_car_speed(self):
        print(f"Car speed: {self.car.speed} km/h")
```

**Comment by JohnScolaro** [comment]
Oh snap, it's my post!

I will note that this comes from my experience cleaning up code from more junior developers. I certainly haven't done any further analysis and looked for instances of this sort of pattern in large mature codebases, so maybe there are some use cases where it makes sense? I don't think I've seen any though.

Very curious to hear other people's thoughts!
